lua/lualine.lua

```diff
@@ -210,7 +204,27 @@ local function lualine_set_theme()
   theme_set = M.options.theme
 end
 
-local function statusline(sections, is_focused)
+local function statusline(is_focused)
+  local sections = nil
+  for _, extension in ipairs(M.extensions) do
+    for _, filetype in ipairs(extension.filetypes) do
+      if vim.bo.filetype == filetype then
```

Comment:
Is it not possible for a filetype to have more then one extention ? currently only the extention that matches at last will be displayed . Maybe we should extend the table instead of replacing it. Maybe we can provide a flag for extention . if the extention sets it to true we can display the extention alongside regular statusline instead of it completely replacing the statusline . What do you think ?

Reply:
> Is it not possible for a filetype to have more then one extention ? 

well my idea for extensions is that a lot of plugins use buffers in which a user does not need a regular statusline but a speciefied one. This is the aim of extensions. I don't see why you would need a lot of extensions for a single filetype in that case. 

> if the extention sets it to true we can display the extention alongside regular statusline instead of it completely replacing the statusline . What do you think ?

I don't like this. At this point this is a plugin component and not an extension

Reply:
Makes sence . I think I naver really got the idea of extentions :)

Reply:
Yeah, maybe the naming is weird. I didn't think too much about it just copied the name from what airline is using for this. If you have any better suggestions then lmk. We can also change this in this pr